moveit_plugins/moveit_simple_controller_manager/include/moveit_simple_controller_manager/action_based_controller_handle.h

```diff
@@ -137,9 +144,12 @@ class ActionBasedControllerHandle : public ActionBasedControllerHandleBase
   bool waitForExecution(const rclcpp::Duration& timeout = rclcpp::Duration(0)) override
   {
     auto result_future = controller_action_client_->async_get_result(current_goal_);
-    if (controller_action_client_ && !done_)
-      return (rclcpp::spin_until_future_complete(node_, result_future, timeout.to_chrono<std::chrono::seconds>()) ==
-              rclcpp::executor::FutureReturnCode::SUCCESS);
+
+    rclcpp::Time start_time = node_->now();
+    auto end_time = start_time + timeout;
```

Comment:
Same here. Also, the default `timeout = rclcpp::Duration(0)` should indicate an infinite timeout